packages/x-data-grid/src/components/base/GridOverlays.tsx

```diff
@@ -32,7 +32,6 @@ const GridOverlayWrapperRoot = styled('div', {
         top: 'var(--DataGrid-headersTotalHeight)',
         left: 0,
         width: 0, // To stay above the content instead of shifting it down
-        height: 0, // To stay above the content instead of shifting it down
```

Comment:
This creates a different issue: https://app.argos-ci.com/mui/mui-x/builds/25849/116364988

I have some ideas around this, I'll take a look